test/specs/bullet.spec.js

```diff
@@ -0,0 +1,307 @@
+        let bulletChart, dataset, containerFixture, f, dataPoint;
+
+        function aTestDataSet() {
+            return new dataBuilder.BulletChartDataBuilder();
+        }
+
+        beforeEach(() => {
+            dataset = aTestDataSet()
+                .withCpuData()
+                .build();
+            bulletChart = chart();
+            dataPoint = dataset[0];
+
+            // DOM Fixture Setup
+            f = jasmine.getFixtures();
+            f.fixturesPath = 'base/test/fixtures/';
+            f.load('testContainer.html');
+
+            containerFixture = d3.select('.test-container');
+            containerFixture.datum(dataPoint).call(bulletChart);
+        });
+
+        afterEach(() => {
+            containerFixture.remove();
+            f = jasmine.getFixtures();
+            f.cleanUp();
+            f.clearCache();
+        });
+
+        describe('when render', () => {
+
+            describe('container groups', () => {
+                it('should render a chart with minimal requirements', () => {
+                    let expected = 1;
+                    let actual = containerFixture.select('.bullet-chart').nodes().length;
+
+                    expect(actual).toEqual(expected);
+                });
+
+                it('should render chart group', () => {
+                    let expected = 1;
+                    let actual = containerFixture.select('.chart-group').nodes().length;
+
+                    expect(actual).toEqual(expected);
+                });
+
+                it('should render axis group', () => {
+                    let expected = 1;
+                    let actual = containerFixture.select('.axis-group').nodes().length;
+
+                    expect(actual).toEqual(expected);
+                });
+
+                it('should render metadata group', () => {
+                    let expected = 1;
+                    let actual = containerFixture.select('.metadata-group').nodes().length;
+
+                    expect(actual).toEqual(expected);
+                });
+
+                it('should render grid lines group', () => {
+                    let expected = 1;
+                    let actual = containerFixture.select('.grid-lines-group').nodes().length;
+
+                    expect(actual).toEqual(expected);
+                });
+            });
+
+            describe('chart components', () => {
+
+                it('should render ranges', () => {
+                    let expected = 3;
+                    let actual = containerFixture.selectAll('rect.range').nodes().length;
+
+                    expect(actual).toEqual(expected);
+                });
+
+                it('should render measures', () => {
+                    let expected = 2;
+                    let actual = containerFixture.selectAll('rect.measure').nodes().length;
+
+                    expect(actual).toEqual(expected);
+                });
+
+                it('should render grid lines group', () => {
```

Comment:
Update message?

Reply:
I will remove this test anyway. For this chart, it would need to have gridlines in `chart-group` since we will have`title` and `subtitle` in `metadata` group.